Packages used in helix-pages are not available for local development.

Packages used in helix-pages are not available for local development.


```
$ mkdir mytest
$ cd mytest/
$ vi index.md
$ git init
$ git add -A
$ git ci -am"initial"
$ hlx up
    __ __    ___       ___
   / // /__ / (_)_ __ / _ \___ ____ ____ ___
  / _  / -_) / /\ \ // ___/ _ `/ _ `/ -_|_-<
 /_//_/\__/_/_//_\_\/_/   \_,_/\_, /\__/___/
                              /___/ v4.2.2

Checking out sources from https://github.com/adobe/helix-pages.git#master
Debugger listening on ws://127.0.0.1:9229/f18d10bb-6c6d-44c1-aa68-f9f100063604
For help, see: https://nodejs.org/en/docs/inspector
Building project files...done 1358ms
[hlx] info: Starting helix-simulator v2.12.12
[hlx] info: Local Helix Dev server up and running: http://localhost:3000/
[git] info: [19559] HTTP: listening on port 54384
[git] error: encountered unhandled promise rejection at process level: [object Promise], reason: Error: Cannot find module 'jquery'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:582:15)
^C
$ npm add jquery
+ jquery@3.4.1

$ hlx up
    __ __    ___       ___
   / // /__ / (_)_ __ / _ \___ ____ ____ ___
  / _  / -_) / /\ \ // ___/ _ `/ _ `/ -_|_-<
 /_//_/\__/_/_//_\_\/_/   \_,_/\_, /\__/___/
                              /___/ v4.2.2

Debugger listening on ws://127.0.0.1:9229/492a0738-739d-4e12-a75c-959d9c79a5da
For help, see: https://nodejs.org/en/docs/inspector
Building project files...done 1149ms
[hlx] info: Starting helix-simulator v2.12.12
[hlx] info: Local Helix Dev server up and running: http://localhost:3000/
[git] info: [19645] HTTP: listening on port 54403
[hlx] error: Could not find Markdown at http://127.0.0.1:54403/raw/helix/localhost--local--default/master/header.md
[hlx] error: Could not find Markdown at http://127.0.0.1:54403/raw/helix/localhost--local--default/master/footer.md
```

--- a/src/helix-pages.js
+++ b/src/helix-pages.js
@@ -12,6 +12,7 @@
 const path = require('path');
 const fse = require('fs-extra');
 const shell = require('shelljs');
+const chalk = require('chalk');
 const { GitUrl } = require('@adobe/helix-shared');
 
 function execAsync(cmd) {
@@ -83,11 +84,26 @@ class HelixPages {
     this._staticURL = new GitUrl('https://github.com/adobe/helix-pages.git/htdocs');
 
     if (!await fse.pathExists(this.checkoutDirectory)) {
-      this.log.info(`Checking out sources from ${this._repo}#${this._ref}`);
+      this.log.info(chalk`Checking out sources from {cyan ${this._repo}#${this._ref}}`);
       try {
         await execAsync(`git clone --branch ${this._ref} --quiet --depth 1 ${this._repo} ${this.checkoutDirectory}`);
       } catch (e) {
         throw Error(`Unable to checkout helix-pages repository: ${e}`);
+      }
+
+      const pagesPackageJson = path.resolve(this.checkoutDirectory, 'package.json');
+      if (await fse.pathExists(pagesPackageJson)) {
+        const pkgJson = await fse.readJson(pagesPackageJson);
+        this.log.info(chalk`Running {gray npm install} for {yellow ${pkgJson.name}@${pkgJson.version}}...`);
+        const cwd = process.cwd();
+        try {
+          shell.cd(this.checkoutDirectory);
+          await execAsync('npm install --only=prod --ignore-scripts --no-bin-links --no-audit');
+        } catch (e) {
+          throw Error(`Unable to install helix-pages dependencies: ${e}`);
+        } finally {
+          shell.cd(cwd);
+        }
       }
     }
   }

--- a/src/up.cmd.js
+++ b/src/up.cmd.js
@@ -205,6 +205,10 @@ class UpCommand extends BuildCommand {
         this.config.strains.get('default').static.url = pages.staticURL;
       }
 
+      // append the pages node_modules to the module path
+      const pagesModules = path.resolve(pages.checkoutDirectory, 'node_modules');
+      this._project.withRuntimeModulePaths([pagesModules, ...module.paths]);
+
       // pretend to have config file to suppress message below
       hasConfigFile = true;
     } else {

--- a/test/fixtures/helix-pages/master/package.json
+++ b/test/fixtures/helix-pages/master/package.json
@@ -1,0 +1,11 @@
+{
+  "name": "helix-pages",
+  "version": "1.0.0",
+  "description": "minimal checkout of helix pages for testing",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": "https://github.com/adobe/helix-pages.git",
+  "license": "Apache-2.0"
+}

